src/sentry.tsx

```diff
@@ -0,0 +1,54 @@
+import * as Sentry from "@sentry/browser";
+import {
+  Integrations as ApmIntegrations,
+  Span,
+  Transaction,
+} from "@sentry/apm";
+import { CaptureConsole as CaptureConsoleIntegration } from "@sentry/integrations";
+
+const Tracing = ApmIntegrations.Tracing;
+
+export const setupSentry = (options: { [key: string]: any }) => {
+  console.log(`[sentry] Initialized with DSN: ${options.dsn}`);
+  Sentry.init({
+    tracesSampleRate: 1.0,
+    ...options,
+    integrations: [
+      new CaptureConsoleIntegration({
+        levels: ["error"],
+      }),
+      new ApmIntegrations.Tracing(),
+    ],
+  });
+};
+
+export const startSpan = (options: {
+  description?: string;
+  op?: string;
+}): Span | undefined => {
+  const tracingIntegration = Sentry.getCurrentHub().getIntegration(Tracing);
+  if (!tracingIntegration) {
+    console.warn("startSpan called without tracing integration");
+    return undefined;
+  }
+  const transaction = (tracingIntegration as any).constructor.getTransaction();
+  if (!transaction) {
+    console.info("startSpan called without transaction");
+    return undefined;
+  }
+  return transaction.startChild(options);
+};
+
+export const startTransaction = (options: {
```

Comment:
this doesnt actually seem to work well.. to some degree i wanted to manually bind a transaction but the tracing integration couples them to itself, and theres no real APIs afaik to manage them without the idle helper

Reply:
Can't you just use `Sentry.startTransaction` https://github.com/EvanPurkhiser/prolink-connect/blob/master/src/network.ts#L102

Reply:
damn - didnt find that one :(